elastic/kibana
# [App Search] Set up Curations routes & complete 'Edit Query' action in Analytics tables

## Summary

This PR is sort of a half-and-half for setting up Curations and finishing Analytics.

- Curations - 7b62f3a & c353437 & 7f8724a
  - Sets up the CurationsRouter (with TODOs)
  - Updates EngineRouter & EngineNav
  - Sets up a single server API route
- Analytics - dbdde94 & 188b931 & d78e8a0
  - Adds missing TODO `onClick` functionality to the Edit Query action button/icon
  - Refactors tests for AnalyticsTable & RecentQueriesTable to call a DRY'd out set of test suites for the actions column (so they should run the same tests each per-file, without duplicating tests)

## QA

- Analytics
  - Go to http://localhost:5601/xyz/app/enterprise_search/app_search/engines/national-parks-demo/analytics
  - Go to a query that doesn't have an existing curation set, e.g. "test"
  - [x] Click the pencil icon and confirm it takes you to a curation ID route
  - [Optional] Go to http://localhost:3002/as#/engines/national-parks-demo/curations and confirm the curation was correctly created
  - [x] Go back to the "test" query and click the pencil icon again, confirm it takes you back to the same curation ID
- Curations routes: confirm the below routes all work as expected
  - [x] http://localhost:5601/xyz/app/enterprise_search/app_search/engines/national-parks-demo/curations
  - [x] http://localhost:5601/xyz/app/enterprise_search/app_search/engines/national-parks-demo/curations/new
  - [x] http://localhost:5601/xyz/app/enterprise_search/app_search/engines/national-parks-demo/curations/test-id
  - [x] http://localhost:5601/xyz/app/enterprise_search/app_search/engines/national-parks-demo/curations/test-id/add_result
  - [x] http://localhost:5601/xyz/app/enterprise_search/app_search/engines/national-parks-demo/curations/test-id/404

### Checklist
